tutorials/plugins/gdnative/files/cpp_example/SConstruct

```diff
@@ -50,10 +50,10 @@ if env['platform'] == "osx":
     env['target_path'] += 'osx/'
     cpp_library += '.osx'
     if env['target'] in ('debug', 'd'):
-        env.Append(CCFLAGS=['-g', '-O2', '-arch', 'x86_64'])
+        env.Append(CCFLAGS=['-g', '-O2', '-arch', 'x86_64', '-std=c++17'])
```

Comment:
`CCFLAGS` are the flags for both the C and C++ compilers, so we can't pass `-std=c++17` to the C compiler (Clang would normally raise an error, but I guess Apple Clang is more lenient if it works for you).

This should be:
```
env.Append(CXXFLAGS=['-std=c++17'])
```
to only affect C++.

Reply:
How about now? Moved common flags from conditionals for osx and linux (like it is for windows).